pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -80,6 +88,10 @@ def distributed_sampler_kwargs(self):
         distributed_sampler_kwargs = dict(num_replicas=(self.num_nodes * self.num_processes), rank=self.global_rank)
         return distributed_sampler_kwargs
 
+    @property
+    def is_single_process_single_device(self):
```

Comment:
should it be lifted to the ParallelPlugin class?
also you didn't define it for DP, there it would have to be set false right?

Reply:
did not set up for DP or lifted to the parent class `ParallelPlugin` as currently it is only used for applying model hook to DDP model.

We could lift to ParallelPlugin, then I guess we need to override it to every subclass, which is fine too. 

>  you didn't define it for DP

DP would not apply such hook.